### PR TITLE
Add a py3_image rule

### PR DESCRIPTION
**Pull request by hwright**
This rule is an analog to `py_image`, only the resulting image runs
Python 3 instead of Python 2.  I'm certainly open to suggestions on
a more elegant way of doing this using attributes on the stock
`py_image` rule, but I'm not sure if there is one.  This was mostly
made by copying the implementation of `py_image`.

This change also updates a bunch of unrelated stuff as a consequence
of running `update_deps.sh`.

This fixes bazelbuild/rules_docker#229

### CODE REVIEW

**Comment by bazel-io** [comment]
Can one of the admins verify this patch?

File: python3/image.bzl [hunk 1]
```diff
@@ -0,0 +1,92 @@
...
+  binary_name = name + ".binary"
+
+  # TODO(mattmoor): Consider using par_binary instead, so that
+  # a single target can be used for all three.
+  native.py_binary(name=binary_name, deps=deps + layers, **kwargs)
+
+  # TODO(mattmoor): Consider what the right way to switch between
+  # Python 2/3 support might be.  Perhaps just overriding `base`,
+  # but perhaps we can be smarter about selecting a py2 vs. py3
+  # distroless base?
```
Comment (mattmoor) [inline python3/image.bzl]: Seems like we're making a call here :)

Can you drop it here and the source?

File: testdata/BUILD [hunk 2]
```diff
@@ -502,6 +502,22 @@ py_image(
     main = "py_image.py",
 )
 
+load("//python3:image.bzl", "py3_image")
+
+py_library(
+    name = "py3_image_library",
+    srcs = ["py3_image_library.py"],
+)
```
Comment (mattmoor) [inline testdata/BUILD]: Not sure this is meaningfully different, can we just use the existing `py_libary`?  LMK if I'm missing a change.

**Comment by mattmoor** [comment]
Test this please

File: testdata/BUILD (hunk 2 above)
Comment (hwright) [reply testdata/BUILD]: Nope, this can be removed and the existing one reused.

File: python3/image.bzl (hunk 1 above)
Comment (hwright) [reply python3/image.bzl]: Done.

Though I'm still on the fence about the approach here: we could logically extend it toward treating `py_binary` and `py_library` as python 2-only, and then adding `py3_binary` and `py3_library` rules to `rules_python` and so forth.  They are two different languages in some senses, though I don't know if I want that precedent here.

Comment (mattmoor) [reply python3/image.bzl]: I will leave that to the Python gods to decide.  I just play one on GitHub :)

@duggelz FYI

**Comment by mattmoor** [comment]
Test this please

**Comment by mattmoor** [comment]
Thanks again @hwright :)

**Comment by mattmoor** [comment]
It occurred to me immediately after pressing merge that you didn't update [this](https://github.com/bazelbuild/rules_docker#language-rules), which will improve discoverability.

**Comment by hwright** [comment]
And I haven't even managed to start playing one on github: #238 :)

**Pull request merged**